readingDataPointsFilters:
  - MaxDistDataPointsFilter:
      dim: -1
      maxDist: 5
  #- MinDistDataPointsFilter:
      #dim: -1
      #minDist: 0.3
  #- RemoveNaNDataPointsFilter
  - RandomSamplingDataPointsFilter:
      prob: 0.35

referenceDataPointsFilters:
  - MaxDistDataPointsFilter:
      dim: -1
      maxDist: 5
  #- MinDistDataPointsFilter:
      #dim: -1
      #minDist: 0.3
  #- RemoveNaNDataPointsFilter
  - RandomSamplingDataPointsFilter:
      prob: 0.35
  #- SamplingSurfaceNormalDataPointsFilter:
      #knn: 10

matcher:
  KDTreeMatcher:
    maxDist: 0.5
    knn: 1
    epsilon: 3.16

outlierFilters:
  - TrimmedDistOutlierFilter:
     ratio: 0.85
#  - SurfaceNormalOutlierFilter:
#     maxAngle: 0.42

errorMinimizer:
  #PointToPlaneWithCovErrorMinimizer
  PointToPointWithCovErrorMinimizer

transformationCheckers:
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.001
      minDiffTransErr: 0.01
      smoothLength: 4
  - CounterTransformationChecker:
      maxIterationCount: 100
  #- BoundTransformationChecker:
      #maxRotationNorm: 0.80
      #maxTranslationNorm: 5.00

inspector:
# VTKFileInspector
 NullInspector

logger:
  NullLogger
#  FileLogger
